components/ee/agent-smith/pkg/agent/agent.go

```diff
@@ -138,6 +143,7 @@ func NewAgentSmith(cfg Config) (*Smith, error) {
 		notifiedInfringements: notificationCache,
 		perfHandler:           make(chan perfHandlerFunc, 10),
 		metrics:               newAgentMetrics(),
+		pidsMap:               sync.Map{},
```

Comment:
nit: this is essentially a noop, because `pidsMap` will already be initialized with the default value